extensions/Khronos/KHR_materials_cmnBlinnPhong/README.md

```diff
@@ -0,0 +1,134 @@
+
+## Extending Materials
+
+The common Blinn-Phong material is defined by adding the `KHR_materials_cmnBlinnPhong` extension to any glTF material. 
+For example, the following defines a material having all possible parameters.:
+
+```
+{
+    "materials": [
+        {
+            "name": "All parameters",
+            "extensions": {
+                "KHR_materials_cmnBlinnPhong": {
+                    "diffuseFactor" : [
+                        0.8, 
+                        0.8, 
+                        0.8, 
+                        1.0
+                    ], 
+                    "diffuseTexture" : {
+                        "index" : 0
+                    }, 
+                    "shininessFactor" : 2.0, 
+                    "specularFactor" : [
+                        0.1, 
+                        0.1, 
+                        0.1
+                    ],
+                    "specularShininessTexture" : {
+                        "index" : 1
+                    }, 
+                }
+            }
+
+        }
+    ]
+}
+```
+### Blinn - Phong
+
+The Blinn-Phong material model is defined by the following properties:
+* `diffuse` - Reflected diffuse color of the material
+* `specular` - Specular color of the material
+* `shininess` - Shininess factor of the specular color
+
+TODO: EXPLAIN FURTHER 
+
+The following table lists the allowed types and ranges for the specular-glossiness model:
+
+|                            |Type         |Description|Required|
+|----------------------------|-------------|-----------|--------|
+|**diffuseFactor**           | `number[4]` | The reflected diffuse RGBA factor of the material.|No, default: `[1.0,1.0,1.0,1.0]`|
+|**diffuseTexture**          | [`textureInfo`](/specification/2.0/README.md#reference-textureInfo) | The diffuse texture.|No|
+|**specularFactor**          | `number[3]` | The specular RGB color of the material.      |No, default: `[1.0,1.0,1.0]`|
+|**shininessFactor**         | `number`    | The shininess of the material.|No, default: `1.0`          |
+|**specularShininessTexture**| [`textureInfo`](/specification/2.0/README.md#reference-textureInfo)|The specular shininess texture.|No|
+
+Additional properties are allowed and may lead to undefined behaviour in conforming viewers.
+
+The material is intended for shading according to the Blinn-Phong lighting model or a close approximation.
+
+This equation is not complex and detailed via the ACM. Refer to “Models of Light
+Reflection for Computer Synthesized Pictures,” SIGGRAPH 77, pp 192-198 [http://portal.acm.org/citation.cfm?id=563893](http://portal.acm.org/citation.cfm?id=563893), and in particular the "Simple Highlight Model" section.
+
+The following code illustrates the basic computation:
+
+```
+color = <emission> + <ambient> * al + <diffuse> * max(N * L, 0) + <specular> * max(H * N, 0)^<shininess>

```

Comment:
where is `<ambient>` defined?

Reply:
on one of the other threads i did suggest adding the following (it is submitted as a PR against what i suspect is now a dead fork)

diffuseTerm   = diffuseFactor   * diffuseTexture 
specularTerm  = specularFactor  * specularShininessTexture.rgb 
shineTerm     = shininessFactor * specularShininessTexture.a
emissiveTerm  = emissiveFactor  * emissiveTexture // core Materials spec
occlusionTerm = occlusionFactor * occlusionTexture // core Materials spec 

color = emissiveTerm + occlusionTerm * (diffuseTerm * max(N * L, 0) + specularTerm * max(H * N, 0)^shineTerm)

where
 `N` – Normal vector
 `L` – Light vector
 `H` – Half-angle vector,calculated as halfway between the unit Eye and Light vectors, using the equation H= normalize(I+L)

Textures are optional, and if provided, will modulate the base Factor value. Note that `specularShininessTexture` contains the specular modulation as the rgb components and shininess modulation as the alpha component.